tutorial/LFRic/background/LFRic_kernel.md

```diff
@@ -0,0 +1,241 @@
+
+```fortran
+use kernel_mod,        only: kernel_type
+```
+
+Looking at the `setval_field_w0_kernel_type` above, there are two main
+parts of an LFRic kernel:
+* Metadata that tells PSyclone how to operate on a kernel;
+* Subroutine with the argument list, declarations and executable code.
+
+All of this is stored in a module. The naming convention for the module,
+the kernel type and the executable subroutine is as follows:
+
+* Module name: `<base_name>_kernel_mod`;
+* Kernel type name: `<base_name>_kernel_type`;
+* Subroutine name: `<base_name>_code`.
+
+Here the `<base_name>` is `setval_field_w0`.
+
+### Kernel metadata
+
+The PSyclone metadata are stored as the *private* data within the definition
+of an individual kernel type. They come in two main forms: derived types
+(e.g. `arg_type` in the above example) and `integer` arguments (single-valued
+such as `iterates_over` above or arrays). Here we briefly explain the contents
+of `arg_type` and `iterates_over` metadata. For more information please refer
+to the [*Metadata* section](
+https://psyclone.readthedocs.io/en/stable/dynamo0p3.html#metadata)
+of the LFRic (Dynamo 0.3) API [user documentation.](
+https://psyclone.readthedocs.io/en/stable/dynamo0p3.html)
+
+The `arg_type` in this example describes the two (hence `dimension(2)`)
+arguments that this kernel operates on:
+* A field argument (`GH_FIELD`) with the degrees of freedom (DoFs) on `W0`
+  function space that is being updated in this kernel (`GH_INC` access for
+  updating fields on continuous function spaces);
+* A `real`-valued scalar (`GH_REAL`) whose value is read (`GH_READ`) to
+  update the field with.
+
+The `GH_` prefix comes from "GungHo", the name of the dynamical core used in
+the LFRic model. Please refer to the documentation for more information on
+[supported accesses](
+https://psyclone.readthedocs.io/en/stable/dynamo0p3.html#valid-access-modes)
+and [function spaces](
+https://psyclone.readthedocs.io/en/stable/dynamo0p3.html#supported-function-spaces)
+for the LFRic data objects.
+
+The `iterates_over` metadata specifies how LFRic kernels are called
+from the [PSy layer](LFRic_PSy.md). A user-defined kernel in LFRic
+is called from a PSy-layer loop over each cell in the horizontal domain,
+hence the metadata identifier for this way of looping, `CELLS`. This means
+that a kernel operates on a one-cell-wide vertical column of cells.
+
+---
+**NOTE**
+
+* LFRic field data are currently `real`-valued, however there is ongoing
+  work to allow the field data of other intrinsic types, such as `integer`.
+  This will be reflected in the metadata, with `GH_REAL` and `GH_INTEGER`
+  marking the data type and `GH_SCALAR` denoting the scalar arguments. The
+  above `arg_type` notation will change to
+
+  ```fortran
+      type(arg_type), dimension(2) :: meta_args = (/ &
+           arg_type(GH_FIELD,  GH_REAL, GH_INC, W0), &
+           arg_type(GH_SCALAR, GH_REAL, GH_READ)     &
+           /)
+  ```
```

Comment:
I understand why you want to include this but is it going to lead to confusion in the reader? Might it be better to leave it out and change it when the implementation changes? Maybe an appendix instead?

Reply:
I pushed it to the appendix.